pkg/types/dsse/dsse.go

```diff
@@ -97,6 +98,14 @@ func verifyEnvelope(allPubKeyBytes map[*pb.Verifier][]byte, pbenv *pbdsse.Envelo
 			return nil, fmt.Errorf("could not parse public key as x509: %w", err)
 		}
 
+		valid, err := types.CheckEntryAlgorithms(key, alg, algorithmRegistry)
+		if err != nil {
+			return nil, fmt.Errorf("checking entry algorithm: %w", err)
+		}
+		if !valid {
+			return nil, fmt.Errorf("invalid entry algorithm")
```

Comment:
include the key,alg in error?

Reply:
I don't have a concise way to refer to the key but I can give the algorithm